CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationRewards.uc

```diff
@@ -277,6 +278,60 @@ static function X2ActivityTemplate_Infiltration GetInfiltrationTemplateFromRewar
 	return X2ActivityTemplate_Infiltration(Activity.GetMyTemplate());
 }
 
+static function X2DataTemplate CreateDarkEventRewardTemplate()
+{
+	local X2RewardTemplate Template;
+	
+	// This is a dummy reward, does nothing
+	`CREATE_X2Reward_TEMPLATE(Template, 'Reward_DarkEvent');
+	
+	Template.SetRewardFn = SetDarkEventReward;
+	Template.GenerateRewardFn = GenerateDarkEventReward;
+	Template.GetRewardStringFn = GetDarkEventRewardString;
+	Template.GetRewardPreviewStringFn = GetDarkEventRewardString;
+
+	return Template;
+}
+
+static function SetDarkEventReward(XComGameState_Reward RewardState, optional StateObjectReference RewardObjectRef, optional int Amount)
+{
+	RewardState.RewardObjectReference = RewardObjectRef;
+}
+
+static function GenerateDarkEventReward (XComGameState_Reward RewardState, XComGameState NewGameState, optional float RewardScalar = 1.0, optional StateObjectReference AuxRef)
+{
+	local XComGameState_ActivityChain ChainState;
+	local XComGameState_MissionSite MissionState;
+	local XComGameState_DarkEvent DarkEventState;
+	local XComGameStateHistory History;
+
+	History = `XCOMHISTORY;
+
+	foreach NewGameState.IterateByClassType(class'XComGameState_MissionSite', MissionState)
+	{
+		if (MissionState != none) break;
+	}
+	
+//[0480.70] ScriptWarning: Accessed None 'StateObjectID'
+	ChainState = class'XComGameState_Activity'.static.GetActivityFromPrimaryObjectID(MissionState.ObjectID).GetActivityChain();
```

Comment:
I'm having some trouble here, I need to get the ChainState from the RewardState but I can't think of a way to do it other than this and as you can see from the error comments it isn't really working.